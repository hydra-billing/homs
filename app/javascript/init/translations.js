import 'i18n-js';
import { en } from '../../../config/locales/en.yml'
import { ru } from '../../../config/locales/ru.yml'

I18n.translations = { en, ru };
