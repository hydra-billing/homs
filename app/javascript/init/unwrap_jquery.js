window.$ = modulejs.require('jQuery');
window.jQuery = $;
