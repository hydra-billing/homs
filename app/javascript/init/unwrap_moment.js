window.moment = modulejs.require('moment');
