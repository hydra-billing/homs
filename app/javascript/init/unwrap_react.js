window.React = modulejs.require('React');
