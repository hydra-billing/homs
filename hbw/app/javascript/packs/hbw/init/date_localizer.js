import { parseISO, format } from 'date-fns';
import en from 'date-fns/locale/en-US';
import ru from 'date-fns/locale/ru';

const locales = { en, ru };

export const localize = (str, locale, dateTimeFormat) => {
  if (str) {
    return format(parseISO(str), dateTimeFormat, { locale });
  }

  return null;
};

const localizeDatetime = (locale, dateTimeFormat) => (
  str => localize(str, locales[locale], dateTimeFormat)
);

const localizeDayMonth = locale => (
  str => format(parseISO(str), 'dd MMM', locales[locale])
);

const localizeDayMonthYear = locale => (
  str => format(parseISO(str), 'dd MMM yyyy', locales[locale])
);

export const localizer = ({ code: locale, dateTimeFormat }) => (
  {
    localizeDatetime:     localizeDatetime(locale, dateTimeFormat),
    localizeDayMonth:     localizeDayMonth(locale),
    localizeDayMonthYear: localizeDayMonthYear(locale)
  }
);
