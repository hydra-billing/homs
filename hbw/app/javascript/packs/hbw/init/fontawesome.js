import { library } from '@fortawesome/fontawesome-svg-core';
import { fab } from '@fortawesome/free-brands-svg-icons';
import { far } from '@fortawesome/free-regular-svg-icons';
import { fas } from '@fortawesome/free-solid-svg-icons';

library.add(fab, far, fas);
