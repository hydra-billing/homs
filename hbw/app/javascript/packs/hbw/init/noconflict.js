const HBWDependencies = ['$', 'jQuery', 'moment', 'React', 'ReactDOM', 'define', 'require'];

window._globalsBeforeHBW = {};

HBWDependencies.forEach((name) => {
  window._globalsBeforeHBW[name] = window[name];

  if (name in window) {
    window[name] = undefined;
  }
});
