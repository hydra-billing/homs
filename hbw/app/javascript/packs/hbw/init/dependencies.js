import './noconflict';
import 'jquery';
import 'moment';
import 'bootstrap-datetimepicker';
import 'tooltip';
import './wrap_globals';
