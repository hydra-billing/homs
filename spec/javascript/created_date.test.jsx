import React from 'react';
import { shallow } from 'enzyme';
import {
  addSeconds, addMinutes, addHours, addDays, addWeeks, addYears,
} from 'date-fns';
import CreatedDate from 'hbw/components/shared/element/created_date';
import Translator from 'hbw/translator';
import { localizer } from 'hbw/init/date_localizer';

describe('CreatedDate should render correctly', () => {
  const fakeEnv = {
    translator: Translator.getTranslatorForLocale('en'),
    localizer:  localizer('en')
  };

  const now = new Date(2019, 11, 2);
  const sec = addSeconds(now, -30);
  const minutes = addMinutes(sec, -2);
  const hours = addHours(minutes, -3);
  const days = addDays(hours, -4);
  const weeks = addWeeks(days, -3);
  const years = addYears(weeks, -1);

  const shallowComponent = date => (
    shallow(<CreatedDate env={fakeEnv} dateISO={date.toISOString()} now={now} />)
  );

  it('with creation date in last year', () => {
    expect(shallowComponent(years).html()).toEqual('06 Nov 2018');
  });

  it('with creation date in last weeks', () => {
    expect(shallowComponent(weeks).html()).toEqual('06 Nov');
  });

  it('with creation date in last days', () => {
    expect(shallowComponent(days).html()).toEqual('4d ago');
  });

  it('with creation date in last hours', () => {
    expect(shallowComponent(hours).html()).toEqual('3h ago');
  });

  it('with creation date in last minutes', () => {
    expect(shallowComponent(minutes).html()).toEqual('2m ago');
  });

  it('with creation date in last seconds', () => {
    expect(shallowComponent(sec).html()).toEqual('&lt;1m ago');
  });
});
