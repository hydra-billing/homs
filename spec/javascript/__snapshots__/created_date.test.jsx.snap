// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CreatedDate should render correctly with creation date in last days 1`] = `
<div>
  4d ago
</div>
`;

exports[`CreatedDate should render correctly with creation date in last hours 1`] = `
<div>
  3h ago
</div>
`;

exports[`CreatedDate should render correctly with creation date in last minutes 1`] = `
<div>
  2m ago
</div>
`;

exports[`CreatedDate should render correctly with creation date in last seconds 1`] = `
<div>
  &lt;1m ago
</div>
`;

exports[`CreatedDate should render correctly with creation date in last weeks 1`] = `
<div>
  06 Nov
</div>
`;

exports[`CreatedDate should render correctly with creation date in last year 1`] = `
<div>
  06 Nov 2018
</div>
`;
